libmattermost.c

```diff
@@ -108,6 +108,22 @@ json_array_to_string(JsonArray *array)
 	return str;
 }
 
+// Mattermost uses unnamed arrays that json-glib parser turns into strings ...
+JsonArray *
+json_named_array_from_string(const gchar *name, const gchar *str)
+{
+	JsonArray *array = NULL;
+	JsonParser *parser = json_parser_new();
+	gchar *data = g_strconcat("{\"", name , "\":", str, "}", NULL);
+

```

Comment:
you are absolutely right ! .. no need to do this hacking: we just need to re-parse the string. code changed (left as separate function since in many event messages it could be useful)